### ISSUE TITLE
Some testsuite?

### ISSUE THREAD

**Issue by pfalcon** [dabeaz/wadze#2]
> Zero Extras

> Smaller is better.

Absolutely great, please keep it up! But some part is however missing - some kind of testsuite. Could be as simple as "smoke test" which just parses included `input.wasm` and checks for expected output.

**Comment by dabeaz** [comment]
Agree some test might be useful.  Mainly just pressed for time at the moment, so doing so is low priority.  There is an official Wasm test-suite of some kind.   Looking into something like that (as possible input for a test) might be interesting.

**Comment by windelbouwman** [comment]
The wasm test suite is organized as S expressions, and there are indeed binary modules defined in those S expressions. The spec test can be found here: https://github.com/WebAssembly/spec/tree/master/test/core

As an example, this is what I did to run the spec test against ppci: https://github.com/windelbouwman/ppci-mirror/blob/master/test/wasm/test_suite_full.py

**Comment by dabeaz** [comment]
Just noting that it is extremely unlikely I will be working on a test-suite for Wadze.  The library is pretty much what it is--a simple Wasm decoder written in a purely functional style.   I will gladly fix bugs in it, but if you're using Wadze, you're probably building something greater.  The burden of testing that (whatever it is) against the official Wasm spec lies with your project.

**Comment by pfalcon** [comment]
I just read the "FAQ" part of https://github.com/dabeaz/typemap#faq :

> Are there any unit tests?
> No, not tests, types.

I guess similar should have been answered here too :-D.

**Comment by dabeaz** [comment]
Perhaps wadze takes it a step further by having no tests or types!

Honestly, this library pretty much is what it is.  I think it's a fairly clean implementation of a Wasm decoder.  Maybe someone else could take this and build something bigger with it.    I might even do that myself some day.  However,  my main work is in teaching so doing so is still a low priority.

**Comment by pfalcon** [comment]
Absolutely, and no worries. Your works are inspiring to many people, myself including, and I find myself returning to them regularly, and find new interesting, and sometimes funny, things ;-). Surely, someone interested in using them, should be prepared to work further on them, not just expect everything to be done already.
